src/rpc/RPC.cpp

```diff
@@ -1,11 +1,46 @@
-#include <boost/asio/spawn.hpp>
 #include <etl/ETLSource.h>
 #include <rpc/Handlers.h>
 #include <rpc/RPCHelpers.h>
+#include <webserver/HttpBase.h>
+#include <webserver/WsBase.h>
+
+#include <boost/asio/spawn.hpp>
+
 #include <unordered_map>
 
 namespace RPC {
 
+Context::Context(
+    boost::asio::yield_context& yield_,
+    std::string const& command_,
+    std::uint32_t version_,
+    boost::json::object const& params_,
+    std::shared_ptr<BackendInterface const> const& backend_,
+    std::shared_ptr<SubscriptionManager> const& subscriptions_,
+    std::shared_ptr<ETLLoadBalancer> const& balancer_,
+    std::shared_ptr<ReportingETL const> const& etl_,
+    std::shared_ptr<WsBase> const& session_,
+    util::TagDecoratorFactory const& tagFactory_,
+    Backend::LedgerRange const& range_,
+    Counters& counters_,
+    std::string const& clientIp_)
+    : Taggable(tagFactory_.make(std::cref(session_->tag())))
```

Comment:
I think `session_` is null when the request comes over regular http and not websocket. Did you test this case?

Reply:
I just checked and you are totally right - nicely spotted! I will have to think about how to workaround the fact we can have an empty session. If i just don't use the session if it's `null` then we will have a random `id` for the context without any coupling to the underlying session be it ws or http.

Reply:
well when session is null, the request is over http, and I think then we can probably just use the same tag as the http session?